### DISCUSSION TITLE
How to schedule tagless replicas only at tagless disks?

### DISCUSSION THREAD

**Discussion by lictw** [longhorn/longhorn#4806] [General]
Hi!

I added new hdd disk and set hdd label on it and now i want that only replicas of volumes with tag hdd can be scheduled on this disk, how can i achieve this without "default" tag on all others disks? Can disk selector of default SC locates to empty tag? Or some setting "dont create replicas without tags on tagged disks"? Btw strange behavior by default, it was unexpected..

**Comment by joshimoo** [comment]
cc @innobead @c3y1huang I think this is currently not possible, since the absence of a selector would allow for scheduling on all nodes can you confirm and convert this to an enhancement issue if so.

**Reply by PhanLe1010** [reply]
The current behavior is like nodeSelector of pod. If there is a node selector, pod can only be scheduled on these node. If pod has no selector, it can be schedule anywhere.

**Comment by grzleadams** [comment]
This seems like a major oversight (implementing node selector-like behavior without taint- and toleration-like behavior). We have some workloads that _need_ to run on certain disks and use tags for those, but would also like to keep everything else (i.e., all untagged volumes) off those disks.

**Reply by grzleadams** [reply]
Just to add, I realize this is now 1.5 years old, but still seems to be lacking as of v1.5.5...

**Reply by grzleadams** [reply]
Of course, just to make me look dumb, it looks like this functionality was [added in 1.6.0](https://longhorn.io/docs/1.6.0/references/settings/#allow-empty-node-selector-volume)...
